pkgs/by-name/sh/shoko/package.nix

```
@@ -0,0 +1,50 @@
+{
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnet-sdk_8,
+  dotnet-aspnetcore_8,
+  nixosTests,
+  lib,
+  mediainfo,
+  rhash,
+}:
+buildDotnetModule (finalAttrs: {
+  pname = "shoko";
+  version = "5.1.0";
+
+  src = fetchFromGitHub {
+    owner = "ShokoAnime";
+    repo = "ShokoServer";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-ZO5S0zMwzr4giaO1bmQ4dLBIPrv6eZY7k9Os4GiO4C4=";
+    fetchSubmodules = true;
+  };
+
+  dotnet-sdk = dotnet-sdk_8;
+  dotnet-runtime = dotnet-aspnetcore_8;
+
+  nugetDeps = ./deps.json;
+  projectFile = "Shoko.CLI/Shoko.CLI.csproj";
+  dotnetBuildFlags = "/p:InformationalVersion=\"channel=stable\"";
+
+  executables = [ "Shoko.CLI" ];
+  makeWrapperArgs = [
+    "--prefix"
+    "PATH"
+    ":"
+    "${mediainfo}/bin"
+  ];
+  runtimeDeps = [ rhash ];
+
+  passthru.tests.shoko = nixosTests.shoko;
```

Since https://github.com/Mic92/nix-update/releases/tag/1.10.0, `nix-update-script` can also update `deps.json`
```suggestion
  passthru = {
    tests.shoko = nixosTests.shoko;
    updateScript = nix-update-script { };
  };
```